### PR TITLE
remove github submodule

### PR DESCRIPTION
**Pull request by samsja**
this PR remove github submodule and replace them with pip install from git.

Additionally it use the official hivemind repo now that the pydantic v2 pr has been merge

### CODE REVIEW

**Comment by samsja** [comment]
related to https://github.com/PrimeIntellect-ai/OpenDiloco/issues/1

File: README.md [hunk 1]
```diff
@@ -30,26 +30,14 @@ source .venv/bin/activate
...
-# Hivemind
-cd hivemind_source
-pip install .
-cp build/lib/hivemind/proto/* hivemind/proto/.
-pip install -e ".[all]"
-cd ..
-# Requirements
-pip install -r requirements.txt
-# Others
-pip install --pre torchdata --index-url https://download.pytorch.org/whl/nightly/cpu
```
Comment (Jackmin801) [inline README.md:42]: torchdata nightly is still required

File: README.md [hunk 2]
```diff
@@ -30,26 +30,14 @@ source .venv/bin/activate
...
-pip install -e ./pydantic_config
-# OpenDiLoCo
 pip install .
 ```
 
 Optionally, you can install flash-attn to use Flash Attention 2.
 This requires your system to have cuda compiler set up.
 ```
 # (Optional) flash-attn
-pip install flash-attn==2.5.8
```
Comment (Jackmin801) [inline README.md:52]: I dont think versions about 2.5.8 work. Theres some weird compiler version / shared library issue. could be wrong tho

**Review by Jackmin801 — CHANGES_REQUESTED** [review]
Nice! added some comments

File: README.md (hunk 1 above)
Comment (samsja) [reply README.md:42]: my bad

Comment (samsja) [reply README.md:42]: fixed

**Review by Jackmin801 — APPROVED** [review]
Nice! lfg

**Pull request merged**